core/model/modx/processors/browser/directory/create.class.php

```diff
@@ -35,7 +35,7 @@ public function initialize() {
             'name' => false,
             'parent' => '',
         ));
-        if (!$this->getProperty('name')) return $this->modx->lexicon('file_err_chmod_ns');
+        if (!strlen($this->getProperty('name'))) return $this->modx->lexicon('file_folder_err_ns_name');
```

Comment:
Have you considered to use a more stricter check?
```suggestion
        if ($this->getProperty('name') === null || $this->getProperty('name') === '') return $this->modx->lexicon('file_folder_err_ns_name');
```

Reply:
strlen will convert the argument into a string for sure, so both empty string and NULL will result in the same behavior anyway.

Casting strlen to boolean will leave only two options. Either a string has something in it (non-0 length), then it is filesystem-specific case starts after, or it was null, empty, whatever, resulting in proper error.

Reply:
You're right!